### PR TITLE
feature/ak-22: Added AWS CloudTrail S3 Bucket plugin, test cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/aws/cloudtrail/cloudtrailS3Bucket.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    category: 'CloudTrail',
+    description: 'Ensure that AWS CloudTrail trail uses the designated Amazon S3 bucket.',
+    more_info: 'Ensure that your Amazon CloudTrail trail is configured to use the appropriated S3 bucket in order to meet regulatory compliance requirements within your organization.',
+    recommended_action: 'Modify ClouTrail trails to configure designated S3 bucket',
+    link: 'https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-update-a-trail-console.html',
+    apis: ['CloudTrail:describeTrails'],
+    settings: {
+        trail_s3_bucket_name: {
+            name: 'Trail S3 Bucket Name',
+            description: 'Amazon S3 bucket name designated for CloudTrail trains',
```
Comment (giorod3) [inline plugins/aws/cloudtrail/cloudtrailS3Bucket.js:15]:
```suggestion
            description: 'Amazon S3 bucket name designated for CloudTrail trails',
```

File: plugins/aws/cloudtrail/cloudtrailS3Bucket.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                return rcb();
+            }
+
+            if (!describeTrails.data.length) {
+                helpers.addResult(results, 2, 'CloudTrail is not enabled', region);
+                return rcb();
+            }
+
+            async.each(describeTrails.data, function(trail, cb){
+                if (!trail.TrailARN || trail.S3BucketName == helpers.CLOUDSPLOIT_EVENTS_BUCKET) return cb();
```
Comment (giorod3) [inline plugins/aws/cloudtrail/cloudtrailS3Bucket.js:51]: check if S3BucketName exists first
```suggestion
                if (!trail.TrailARN || (trail.S3BucketName && trail.S3BucketName == helpers.CLOUDSPLOIT_EVENTS_BUCKET)) return cb();
```

**Review by giorod3 — CHANGES_REQUESTED** [review]
a couple of comments here

File: plugins/aws/cloudtrail/cloudtrailS3Bucket.js (hunk 1 above)
Comment (AkhtarAmir) [reply plugins/aws/cloudtrail/cloudtrailS3Bucket.js:15]: sorry about that :D

**Review by dipsubha06 — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**